prymer/primer3/primer3.py

```diff
@@ -399,55 +380,32 @@ def design(self, design_input: Primer3Input) -> Primer3Result:  # noqa: C901
-        primer3_results: dict[str, str] = {}  # key-value pairs of results reported by Primer3
-
-        def primer3_error(message: str) -> None:
-            """Formats the Primer3 error and raises a ValueError."""
-            error_message = f"{message}: "
-            # add in any reported PRIMER_ERROR
-            if "PRIMER_ERROR" in primer3_results:
-                error_message += primer3_results["PRIMER_ERROR"]
-            # add in any error lines
-            if len(error_lines) > 0:
-                error_message += "\n".join(f"\t\t{e}" for e in error_lines)
-            # raise the exception now
-            raise ValueError(error_message)
-
-        while True:
-            # Get the next line.  Since we want to distinguish between empty lines, which we ignore,
-            # and the end-of-file, which is just an empty string, check for an empty string before
-            # stripping the line of any trailing newline or carriage return characters.
-            line: str = self._subprocess.stdout.readline()
-            if line == "":  # EOF
-                primer3_error("Primer3 exited prematurely")
-            line = line.rstrip("\r\n")
-
-            if line == "=":  # stop when we find the line just "="
-                break
-            elif line == "":  # ignore empty lines
-                continue
-            elif "=" not in line:  # error lines do not have the equals character in them, usually
-                error_lines.append(line)
-            else:  # parse and store the result
-                key, value = line.split("=", maxsplit=1)
-                # Because Primer3 will emit both the input given and the output generated, we
-                # discard the input that is echo'ed back by looking for tags (keys)
-                # that do not match any Primer3InputTag
-                if not any(key == item.value for item in Primer3InputTag):
-                    primer3_results[key] = value
+            if tag.is_sequence_arg:
+                seq_args[tag] = value
+            else:
+                global_args[tag] = value
+        design_primers_retval = primer3.design_primers(seq_args=seq_args, global_args=global_args)
+        primer3_results: dict[str, Any] = {}  # key-value pairs of results reported by Primer3
+
+        for key, value in design_primers_retval.items():
+            # Because Primer3 will emit both the input given and the output generated, we
+            # discard the input that is echo'ed back by looking for tags (keys)
+            # that do not match any Primer3InputTag
+            if key not in Primer3InputTag:
+                primer3_results[key] = value
 
         # Check for any errors.  Typically, these are in error_lines, but also the results can
         # contain the PRIMER_ERROR key.
-        if "PRIMER_ERROR" in primer3_results or len(error_lines) > 0:
-            primer3_error("Primer3 failed")
+        if "PRIMER_ERROR" in primer3_results:
```

Comment:
I'm not sure I follow what's happening here. `error_lines` is no longer present, could be removed from the comment on L402-403. Is the `else` on L407-408 unreachable?

Reply:
Fixed, silly copy and paste left an extra outer `if`.  Good catch!